code/game/machinery/prisongate.dm

```diff
@@ -0,0 +1,59 @@
+/obj/machinery/prisongate
+    name = "prison gate scanner"
+    desc = "A hardlight gate with an ID scanner attached to the side. Good at deterring even the most persistent temporalily embarrassed employee."
+    icon = 'icons/obj/machines/implantchair.dmi'
+    icon_state = "hypnochair"
+    base_icon_state = "hypnochair"
+    use_power = IDLE_POWER_USE
+    power_channel = AREA_USAGE_EQUIP
+    idle_power_usage = 5
+    active_power_usage = 30
+    anchored = TRUE
+    var/gate_active = TRUE
+
+/obj/machinery/prisongate/power_change()
+    . = ..()
+    if(!powered())
+        turn_off()
+    if(powered())
+        turn_on()
+
+/obj/machinery/prisongate/proc/turn_off()
+    gate_active = FALSE
+
+/obj/machinery/prisongate/proc/turn_on()
+    gate_active = TRUE
```

Comment:
these procs only set a variable, and are only used here? why use procs in that case, and just set the variable in power_change? and with that in mind, why not just set the variable to the return value of `powered()`?